pipe/section/section_impls/sqlite_connector/src/lib.rs

```diff
@@ -107,11 +107,24 @@ pub fn escape_table_name(name: impl AsRef<str>) -> String {
 /// generate create table command for provided record batch
 pub fn generate_schema(table_name: &str, df: &dyn DataFrame) -> String {
     let name = escape_table_name(table_name);
+    let columns = generate_column_names(df);
+    format!("CREATE TABLE IF NOT EXISTS \"{name}\" ({columns})",)
+}
+
+pub fn generate_column_names(df: &dyn DataFrame) -> String {
     let columns = df
         .columns()
         .iter()
         .map(|col| col.name())
+        .map(|name| {
+            // escape all quotes by replacing with a double quote
+            name.replace('\"', "\"\"")
+        })
+        .map(|name| {
+            // wrap the name in quotes
+            format!("\"{}\"", name)
+        })
         .collect::<Vec<_>>()
         .join(",");
-    format!("CREATE TABLE IF NOT EXISTS \"{name}\" ({columns})",)
+    columns
```

Comment:
no blocking nits:
- columns variable is not required
- double map can be flattened